### PR TITLE
feat(plugin-conversation): bulk activities fetch + cluster

### PR DESCRIPTION
**Pull request by yjlo**
# Pull Request Template

## Description
Bulk fetch of activities using a list of URLs and cluster url. For future breadcrumb support so that we can bulk fetch activities based on a specific cluster

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Test Coverage

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] added 2 comments to conversation 1: confirm they were added
- [x] added 4 comments to conversation 2: confirm they were added
- bulk fetch all **post** activities in conversation 1
 - [x] assert they equal the result of listActivities
- bulk fetch all **post** activities in conversation 1 and comments 3 and 4 from conversation 2
 - [x] assert they equal the result of listActivities on conversation 1 and comment 3 + 4 in conversation 2
- [x] bulk fetch a activity URL that doesn't exist ---> [] 

**Test Configuration**:
* SDK Version
* Node/Browser Version
* NPM Version

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
 - **Currently failing 11 local tests that have to do with sharing, verbs, and download**
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js [hunk 1]
```diff
@@ -249,6 +249,42 @@ const Conversation = WebexPlugin.extend({
...
+  /**
+  * Gets an array of activities with an array of activity URLS
+  * @param {Array} activties
+  * @param {String} cluster
+  * @returns {Promise<Object>} Resolves with the activities
+  * TODO: add cluster functionality when clusters are ready
+  */
+  bulk_activities_fetch(activties,cluster) {
+    if (!cluster) {
+      const activityUrls = map(activties, 'flag-item');
```
Comment (mwegman) [inline packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js]: Remove this line. This was specifically for flags

File: packages/node_modules/@webex/internal-plugin-conversation/test/integration/spec/get.js [hunk 2]
```diff
@@ -293,5 +293,110 @@ describe('plugin-conversation', function () {
...
+            });
+            // gets activity urls of only comment 3 and 4 from convo_2
+            webex3.internal.conversation.listActivities({conversationId: convo_2.id})
+              .then((convo_2_activities) => {
+                [3,4].map((i) => activityURLs.push(convo_2_activities[i.url]))
+              });
+            webex3.internal.conversation.bulk_activities_fetch(activityURLs)
+              .then((bulk_fetched_activities) => {
+                let expectedActivities = convo_1_activities;
+                expectedActivitie.push(onvo_1_activities[3]);
```
Comment (mwegman) [inline packages/node_modules/@webex/internal-plugin-conversation/test/integration/spec/get.js]: Misspelled `expectedActivities`

Comment (yjlo) [reply packages/node_modules/@webex/internal-plugin-conversation/test/integration/spec/get.js]: weird it pushed the old save

File: packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js [hunk 3]
```diff
@@ -249,6 +249,43 @@ const Conversation = WebexPlugin.extend({
...
+        body: {
+          activityUrls
+        }
+      };
+
+      return this.webex.request(params)
+        .then((res) => {
+          const activitiesArr = [];
+
+          res.body.multistatus.forEach((statusData) => {
```
Comment (chrisadubois) [inline packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js]: hmmmm is multistatus ever not defined? -- we should probably be checking body?.multistyatus

File: packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js [hunk 4]
```diff
@@ -249,6 +249,43 @@ const Conversation = WebexPlugin.extend({
...
+            if (statusData.status === '200') {
+              activitiesArr.push(statusData.data.activity);
```
Comment (chrisadubois) [inline packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js]: same here

File: packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js (hunk 3 above)
Comment (mwegman) [reply packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js]: Not for this particular call but it depends on how defensive we should be against the server. How much do you trust the convo service to always return what they say they will.

File: packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js [hunk 5]
```diff
@@ -249,6 +249,43 @@ const Conversation = WebexPlugin.extend({
...
 
+
+  /**
+  * Gets an array of activities with an array of activity URLS
+  * @param {Array} activties
+  * @param {String} cluster
+  * @returns {Promise<Object>} Resolves with the activities
+  * TODO: add cluster functionality when clusters are ready
+  */
+  bulk_activities_fetch(activties, cluster) {
```
Comment (adamweeks) [inline packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js]: Can we change this to `bulkActivitiesFetch` to fit the naming of the rest of the methods in the plugin?

File: packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js [hunk 6]
```diff
@@ -249,6 +249,43 @@ const Conversation = WebexPlugin.extend({
     return activity;
   },
 
+
+  /**
+  * Gets an array of activities with an array of activity URLS
+  * @param {Array} activties
```
Comment (adamweeks) [inline packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js]: Spelling of `activities`

File: packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js (hunk 3 above)
Comment (adamweeks) [reply packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js]: We should definitely check for `multistatus` to exist.

**Comment by adamweeks** [comment]
Just noticed there are eslint errors as well that need to be corrected.

File: packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js (hunk 5 above)
Comment (yjlo) [reply packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js]: yes sure will change that

File: packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js (hunk 6 above)
Comment (yjlo) [reply packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js]: I fixed that and it keeps being wrong. I'll try again

File: packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js (hunk 3 above)
Comment (yjlo) [reply packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js]: I can do that, doesn't hurt to check if it exists

File: packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js (hunk 4 above)
Comment (yjlo) [reply packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js]: You want to check if the fields exists?

File: packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js (hunk 3 above)
Comment (adamweeks) [reply packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js]: @yjlo correct, just a `if (res.body.multistatus)`.

File: packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js (hunk 4 above)
Comment (adamweeks) [reply packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js]: yes, so something like `if (statusData.status === '200' && statusData.data && statusData.data.activity)`

File: packages/node_modules/@webex/internal-plugin-conversation/test/integration/spec/get.js [hunk 7]
```diff
@@ -293,5 +293,115 @@ describe('plugin-conversation', function () {
             assert.equal(mentions[0].url, activity.url);
           })));
     });
+
+    // TODO: add testing for bulk_activities_fetch() with clusters later
+    describe('#bulk_activities_fetch()', () => {
```
Comment (adamweeks) [inline packages/node_modules/@webex/internal-plugin-conversation/test/integration/spec/get.js]: This needs to be updated to the `bulkActivitiesFetch` method name

File: packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js (hunk 4 above)
Comment (yjlo) [reply packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js]: okay I see

File: packages/node_modules/@webex/internal-plugin-conversation/test/integration/spec/get.js (hunk 7 above)
Comment (yjlo) [reply packages/node_modules/@webex/internal-plugin-conversation/test/integration/spec/get.js]: Sorry I did that on another commit was having trouble getting the tests to run

**Pull request closed without merging**